x/oracle/module.go

```diff
@@ -203,155 +203,160 @@ func (am AppModule) EndBlock(ctx sdk.Context, _ abci.RequestEndBlock) []abci.Val
-				)
-			}
+				minReportedPerWindow := am.keeper.GetMinReportedPerWindow(ctx)
 
-			minHeight := reportedInfo.StartHeight + am.keeper.GetReportedRoundsWindow(ctx)
-			maxMissed := am.keeper.GetReportedRoundsWindow(ctx) - minReportedPerWindow
-			// if we are past the minimum height and the validator has missed too many rounds reporting prices, punish them
-			if height > minHeight && reportedInfo.MissedRoundsCounter > maxMissed {
-				consAddr, err := sdk.ConsAddressFromBech32(validator)
-				if err != nil {
-					panic("invalid consAddr string")
-				}
-				operator := am.keeper.ValidatorByConsAddr(ctx, consAddr)
-				if operator != nil && !operator.IsJailed() {
-					// missing rounds confirmed: slash and jail the validator
-					coinsBurned := am.keeper.SlashWithInfractionReason(ctx, consAddr, height, power.Int64(), am.keeper.GetSlashFractionMiss(ctx), stakingtypes.Infraction_INFRACTION_UNSPECIFIED)
+				if missed {
 					ctx.EventManager().EmitEvent(
 						sdk.NewEvent(
-							types.EventTypeOracleSlash,
+							types.EventTypeOracleLiveness,
 							sdk.NewAttribute(types.AttributeKeyValidatorKey, validator),
-							sdk.NewAttribute(types.AttributeKeyPower, fmt.Sprintf("%d", power)),
-							sdk.NewAttribute(types.AttributeKeyReason, types.AttributeValueMissingReportPrice),
-							sdk.NewAttribute(types.AttributeKeyJailed, validator),
-							sdk.NewAttribute(types.AttributeKeyBurnedCoins, coinsBurned.String()),
+							sdk.NewAttribute(types.AttributeKeyMissedRounds, fmt.Sprintf("%d", reportedInfo.MissedRoundsCounter)),
+							sdk.NewAttribute(types.AttributeKeyHeight, fmt.Sprintf("%d", height)),
 						),
 					)
-					am.keeper.Jail(ctx, consAddr)
-					jailUntil := ctx.BlockHeader().Time.Add(am.keeper.GetMissJailDuration(ctx))
-					am.keeper.JailUntil(ctx, consAddr, jailUntil)
-
-					// We need to reset the counter & array so that the validator won't be immediately slashed for miss report info upon rebonding.
-					reportedInfo.MissedRoundsCounter = 0
-					reportedInfo.IndexOffset = 0
-					am.keeper.ClearValidatorMissedRoundBitArray(ctx, validator)
 
-					logger.Info(
-						"slashing and jailing validator due to liveness fault",
-						"height", height,
-						"validator", consAddr.String(),
-						"min_height", minHeight,
-						"threshold", minReportedPerWindow,
-						"slashed", am.keeper.GetSlashFractionMiss(ctx).String(),
-						"jailed_until", jailUntil,
-					)
-				} else {
-					// validator was (a) not found or (b) already jailed so we do not slash
-					logger.Info(
-						"validator would have been slashed for too many missed repoerting price, but was either not found in store or already jailed",
+					logger.Debug(
+						"absent validator",
+						"height", ctx.BlockHeight(),
 						"validator", validator,
+						"missed", reportedInfo.MissedRoundsCounter,
+						"threshold", minReportedPerWindow,
 					)
 				}
+
+				minHeight := reportedInfo.StartHeight + am.keeper.GetReportedRoundsWindow(ctx)
```

Comment:
set `reportedRoundsWindow := am.keeper.GetReportedRoundsWindow(ctx)` to not call repeatedly.

Reply:
updated